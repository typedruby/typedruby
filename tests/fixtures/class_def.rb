class A
end

class ::B
  class C
  end
end

class ::B::C::D
end

def test1 => nil
  A.new
end

def test2 => nil
  B::C.new
end

def test3 => nil
  B::C::D.new
end
