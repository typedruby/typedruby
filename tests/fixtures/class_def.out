info: Typechecking...
error: Could not match types:

        @ __ROOT__/tests/fixtures/class_def.rb:12
     12 |  def test1 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/class_def.rb:13
     13 |    A.new
               ^^^ A

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/class_def.rb:12
     12 |  def test1 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/class_def.rb:13
     13 |    A.new
             ^^^^^ A

error: Could not match types:

        @ __ROOT__/tests/fixtures/class_def.rb:16
     16 |  def test2 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/class_def.rb:17
     17 |    B::C.new
                  ^^^ B::C

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/class_def.rb:16
     16 |  def test2 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/class_def.rb:17
     17 |    B::C.new
             ^^^^^^^^ B::C

error: Could not match types:

        @ __ROOT__/tests/fixtures/class_def.rb:20
     20 |  def test3 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/class_def.rb:21
     21 |    B::C::D.new
                     ^^^ B::C::D

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/class_def.rb:20
     20 |  def test3 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/class_def.rb:21
     21 |    B::C::D.new
             ^^^^^^^^^^^ B::C::D
